testinfra/modules/interface.py

```diff
@@ -74,6 +119,37 @@ def addresses(self):
                 addrs.append(splitted[1].split("/", 1)[0])
         return addrs
 
+    @property
+    def is_up(self):
+
+        # Check Connection Carrier Status
+        if int(self.check_output(
+                "cat /sys/class/net/%s/carrier", self.name)) == 1:
+
+            # Check Operator Status
+            if str(self.check_output(
+                    "cat /sys/class/net/%s/operstate", self.name)) == "up":
+                return True
+        else:
+            return False
+
+    @property
+    def mtu(self):
+        return int(self.check_output(
+            "cat /sys/class/net/%s/mtu", self.name))
+
+    @property
+    def is_bridge(self):
+        return self.run_expect([0, 1, 255],
+                               "ip link show %s type bridge",
+                               self.name).rc == 0
```

Comment:
It seems that you could just do `os.path.isdir("/sys/class/net/%s/bridge" % self.name` here.